aspnetcore/fundamentals/servers/aspnet-core-module.md

```diff
@@ -4,38 +4,107 @@ author: rick-anderson
 
+::: moniker-end
+
 Supported Windows versions:
 
 * Windows 7 or later
 * Windows Server 2008 R2 or later
 
-The ASP.NET Core Module only works with Kestrel. The module is incompatible with [HTTP.sys](xref:fundamentals/servers/httpsys) (formerly called [WebListener](xref:fundamentals/servers/weblistener)).
+::: moniker range=">= aspnetcore-2.2"
+
+When hosting out-of-process, the module only works with Kestrel. The module is incompatible with [HTTP.sys](xref:fundamentals/servers/httpsys) (formerly called [WebListener](xref:fundamentals/servers/weblistener)).
+
+::: moniker-end
+
+::: moniker range="< aspnetcore-2.2"
+
+The module only works with Kestrel. The module is incompatible with [HTTP.sys](xref:fundamentals/servers/httpsys) (formerly called [WebListener](xref:fundamentals/servers/weblistener)).
+
+::: moniker-end
 
 ## ASP.NET Core Module description
 
-The ASP.NET Core Module is a native IIS module that plugs into the IIS pipeline to redirect web requests to backend ASP.NET Core apps. Many native modules, such as Windows Authentication, remain active. To learn more about IIS modules active with the module, see [IIS modules](xref:host-and-deploy/iis/modules).
+::: moniker range=">= aspnetcore-2.2"
+
+The ASP.NET Core Module is a native IIS module that plugs into the IIS pipeline to either:
+
+* Host an ASP.NET Core app inside of the IIS worker process (`w3wp.exe`), called the [in-process hosting model](#in-process-hosting-model).
+
+* Redirect web requests to a backend ASP.NET Core app running the [Kestrel server](xref:fundamentals/servers/kestrel), called the [out-of-process hosting model](#out-of-process-hosting-model).
+
+### In-process hosting model
+
+Using in-process hosting, an ASP.NET Core app runs in the same process as its IIS worker process. This removes the performance penalty of proxying requests over the loopback adapter when using the out-of-process hosting model. IIS handles process management with the [Windows Process Activation Service (WAS)](/iis/manage/provisioning-and-managing-iis/features-of-the-windows-process-activation-service-was).
+
+The ASP.NET Core Module:
+
+* Performs app initialization.
+  * Loads the [CoreCLR](/dotnet/standard/glossary#coreclr).
+  * Calls `Program.Main`.
+* Handles the lifetime of the IIS native request.
+
+The following diagram illustrates the relationship between IIS, the ASP.NET Core Module, and an app hosted in-process:
+
+![ASP.NET Core Module](aspnet-core-module/_static/ancm-inprocess.png)
+
+A request arrives from the web to the kernel-mode HTTP.sys driver. The driver routes the request to IIS on the website's configured port, usually 80 (HTTP) or 443 (HTTPS). The module manages the native request context and converts it into a managed request <xref:Microsoft.AspNetCore.Http.HttpContext> for a custom <xref:Microsoft.AspNetCore.Hosting.Server.IServer> implementation, `IISHttpServer`.
```

Comment:
~~custom~~

Reply:
The module receives the native request and passes control to IISHttpServer, which is what converts that request from native to managed.